torchvision/models/detection/ssd.py

```diff
@@ -164,27 +164,39 @@ class SSD(nn.Module):
         'proposal_matcher': det_utils.Matcher,
     }
 
-    def __init__(self, backbone: nn.Module, anchor_generator: DefaultBoxGenerator,
-                 size: Tuple[int, int], num_classes: int,
-                 image_mean: Optional[List[float]] = None, image_std: Optional[List[float]] = None,
-                 head: Optional[nn.Module] = None,
-                 score_thresh: float = 0.01,
-                 nms_thresh: float = 0.45,
-                 detections_per_img: int = 200,
-                 iou_thresh: float = 0.5,
-                 topk_candidates: int = 400,
-                 positive_fraction: float = 0.25):
-        super().__init__()
+    def __init__(
+        self,
+        backbone: nn.Module,
+        anchor_generator: DefaultBoxGenerator,
+        size: Tuple[int, int],
+        num_classes: int,
+        image_mean: Optional[List[float]] = None,
+        image_std: Optional[List[float]] = None,
+        head: Optional[nn.Module] = None,
+        score_thresh: float = 0.01,
+        nms_thresh: float = 0.45,
+        detections_per_img: int = 200,
+        iou_thresh: float = 0.5,
+        topk_candidates: int = 400,
+        positive_fraction: float = 0.25
+    ) -> None:
 
+        super().__init__()
         self.backbone = backbone
-
         self.anchor_generator = anchor_generator
-
         self.box_coder = det_utils.BoxCoder(weights=(10., 10., 5., 5.))
 
         if head is None:
             if hasattr(backbone, 'out_channels'):
-                out_channels = backbone.out_channels
+                # Other models use int as out_channel, we need compatibility.
+                # Also out_channels needs a runtime cast to int for mypy.
+                out_channels_calc = cast(int, backbone.out_channels)
+                if isinstance(out_channels_calc, int):
+                    out_channels = [out_channels_calc]
+                elif isinstance(out_channels_calc, list):
+                    out_channels = out_channels_calc
+                else:
+                    raise ValueError("Backbone should have out_channels either int or List[int] ")
             else:
                 out_channels = det_utils.retrieve_out_channels(backbone, size)
 
```

Comment:
The below assert is bit ugly.
We could instead 
`raise ValueError("Length Specified output_channels for backbone = x, while the length of aspect ratios of Achor generator are y. Which do not match.")`

Reply:
Agreed. Plain `assert`'s should be avoided. Feel free to send a follow-up PR.

Reply:
I will open an issue later to track these followups, there will be many I think